pysb/simulator/stochkit.py

```diff
@@ -0,0 +1,305 @@
+        if algorithm is None:
+            raise SimulatorException("No StochKit algorithm selected")
+        if not re.match(r'\w', algorithm):
+            # Security check failure
+            raise SimulatorException("StochKit algorithm name should contain "
+                                     "only alphanumeric and underscore "
+                                     "characters")
+
+        if algorithm not in ['ssa', 'tau_leaping']:
+            raise SimulatorException(
+                "algorithm must be 'ssa' or 'tau_leaping'")
+
+        executable = get_path('stochkit_{}'.format(algorithm))
+
+        # Output model file to directory
+        fname = os.path.join(self._outdir, 'pysb.xml')
+
+        trajectories = []
+        for i in range(len(self.initials)):
+            # We write all StochKit output files to a temporary folder
+            prefix_outdir = os.path.join(self._outdir, 'output_{}'.format(i))
+
+            # Export model file
+            with open(fname, 'w') as f:
+                f.write(StochKitExporter(self._model).export(
+                    self.initials[i], self.param_values[i]))
+
+            # Assemble the argument list
+            args = '--model {} --out-dir {} -t {:f} -i {:d}'.format(
+                fname, prefix_outdir, t, t_length)
+
+            # If we are using local mode, shell out and run StochKit
+            # (SSA or Tau-leaping or ODE)
+            cmd = '{} {} {}'.format(executable, args, extra_args)
+            self._logger.debug("StochKit run {} of {} (cmd: {})".format(
+                (i + 1), len(self.initials), cmd))
+
+            # Execute
+            try:
+                handle = subprocess.Popen(cmd, stdout=subprocess.PIPE,
+                                          stderr=subprocess.PIPE, shell=True)
+                return_code = handle.wait()
+            except OSError as e:
+                raise SimulatorException("StochKit execution failed: \
+                {0}\n{1}".format(cmd, e))
+
+            try:
+                stderr = handle.stderr.read()
+            except Exception as e:
+                stderr = 'Error reading stderr: {0}'.format(e)
+            try:
+                stdout = handle.stdout.read()
+            except Exception as e:
+                stdout = 'Error reading stdout: {0}'.format(e)
+
+            if return_code != 0:
+                raise SimulatorException("Solver execution failed: \
+                '{0}' output:\nSTDOUT:\n{1}\nSTDERR:\n{2}".format(
+                    cmd, stdout, stderr))
+
+            traj_dir = os.path.join(prefix_outdir, 'trajectories')
+            try:
+                trajectories.extend([np.loadtxt(os.path.join(
+                    traj_dir, f)) for f in sorted(os.listdir(traj_dir))])
+            except Exception as e:
+                raise SimulatorException(
+                    "Error using solver.get_trajectories('{0}'): {1}".format(
```

Comment:
I think this error message needs some tweaking -- I can't figure out what `get_trajectories` is referring to.